frontend/app/src/App.tsx

```diff
@@ -1093,35 +1093,34 @@ export class App extends PureComponent<Props, State> {
       status ===
         ForwardMsg.ScriptFinishedStatus.FINISHED_FRAGMENT_RUN_SUCCESSFULLY
     ) {
-      const successful =
-        status === ForwardMsg.ScriptFinishedStatus.FINISHED_SUCCESSFULLY ||
-        status ===
-          ForwardMsg.ScriptFinishedStatus.FINISHED_FRAGMENT_RUN_SUCCESSFULLY
-
       window.setTimeout(() => {
         // Notify any subscribers of this event (and do it on the next cycle of
         // the event loop)
         this.state.scriptFinishedHandlers.map(handler => handler())
       }, 0)
 
-      if (successful) {
-        // Clear any stale elements left over from the previous run.
-        // (We don't do this if our script had a compilation error and didn't
-        // finish successfully.)
-        this.setState(
-          ({ scriptRunId, fragmentIdsThisRun }) => ({
-            // Apply any pending elements that haven't been applied.
-            elements: this.pendingElementsBuffer.clearStaleNodes(
-              scriptRunId,
-              fragmentIdsThisRun
-            ),
-          }),
-          () => {
-            // We now have no pending elements.
-            this.pendingElementsBuffer = this.state.elements
-          }
-        )
+      // Clear any stale elements left over from the previous run.
+      // (We don't do this if our script had a compilation error and didn't
+      // finish successfully.)
```

Comment:
Is this comment still correct, since we are no longer checking for `successful` here?

Reply:
And with this change we are probably clearing stale elements if there is a compilation error, is this fine or does that have any negative side effects (e.g. during development)?

Reply:
It just barely gets cut off in the diff, but this code only runs if the script completion status is one of `FINISHED_SUCCESSFULLY`, `FINISHED_EARLY_FOR_RERUN`, or `FINISHED_FRAGMENT_RUN_SUCCESSFULLY`.

I think that the comment is still relevant (previously, the `success` variable name was a bit of a misnomer since I'd think that `FINISHED_EARLY_FOR_RERUN` is still a success case), and nothing changes in the case where there's a script compilation error (I double checked manually that the behavior is the same between this branch and `develop`)

Reply:
Sounds good 👍